moped-editor/src/views/projects/projectView/ProjectComponents/ComponentForm.js

```diff
@@ -68,8 +41,17 @@ const ComponentForm = ({
   const doesInitialValueHaveSubcomponents =
     initialFormValues?.subcomponents.length > 0;
 
-  const { register, handleSubmit, control, watch, setValue } = useForm({
+  const {
+    register,
+    handleSubmit,
+    control,
+    watch,
+    setValue,
+    formState: { isValid },
+  } = useForm({
     defaultValues: defaultFormValues,
+    mode: "onChange",
```

Comment:
This mode causes validation with the yup schema each time a form value changes. There is a warning in the [docs](https://react-hook-form.com/api/useform) that I quoted below, but we can take another look if/when this forms get larger or more complex. It is key in keeping the button disabled until valid inputs are entered for now.

> Validation will trigger on the change event with each input, and lead to multiple re-renders. Warning: this often comes with a significant impact on performance.

Reply:
TIL `onChange` mode.  Thanks for the heads-up. I did not notice any sluggishness when testing our humble form.